modules/system/assets/js/snowboard/ajax/Request.js

```diff
@@ -0,0 +1,789 @@
+            if (this.options.handleConfirmMessage.apply(this, [this.confirm]) === false) {
+                return false;
+            }
+
+            return true;
+        }
+
+        // If no plugins have customised the confirmation, use a simple browser confirmation.
+        if (this.snowboard.listensToEvent('ajaxConfirmMessage').length === 0) {
+            return window.confirm(this.confirm);
+        }
+
+        // Run custom plugin confirmations
+        const promises = this.snowboard.globalPromiseEvent('ajaxConfirmMessage', this.confirm, this);
+
+        try {
+            const fulfilled = await promises;
+            if (fulfilled) {
+                return true;
+            }
+        } catch (e) {
+            return false;
+        }
+
+        return false;
+    }
+
+    get form() {
+        if (this.options.form) {
+            return this.options.form;
+        }
+        if (!this.element) {
+            return null;
+        }
+        if (this.element.tagName === 'FORM') {
+            return this.element;
+        }
+
+        return this.element.closest('form');
+    }
+
+    get context() {
+        return {
+            handler: this.handler,
+            options: this.options,
+        };
+    }
+
+    get headers() {
+        const headers = {
+            'X-Requested-With': 'XMLHttpRequest', // Keeps compatibility with jQuery AJAX
+            'X-WINTER-REQUEST-HANDLER': this.handler,
+            'X-WINTER-REQUEST-PARTIALS': this.extractPartials(this.options.update || []),
+        };
+
+        if (this.flash) {
+            headers['X-WINTER-REQUEST-FLASH'] = 1;
+        }
+
+        if (this.xsrfToken) {
+            headers['X-XSRF-TOKEN'] = this.xsrfToken;
+        }
+
+        return headers;
+    }
+
+    get loading() {
+        return this.options.loading || false;
+    }
+
+    get url() {
+        return this.options.url || window.location.href;
+    }
+
+    get redirect() {
+        return (this.options.redirect && this.options.redirect.length) ? this.options.redirect : null;
+    }
+
+    get flash() {
+        return this.options.flash || false;
+    }
+
+    get files() {
+        if (this.options.files === true) {
+            if (FormData === undefined) {
+                this.snowboard.debug('This browser does not support file uploads');
+                return false;
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    get xsrfToken() {
+        let cookieValue = null;
+
+        if (document.cookie && document.cookie !== '') {
+            const cookies = document.cookie.split(';');
+
+            for (let i = 0; i < cookies.length; i += 1) {
+                const cookie = cookies[i].trim();
+
+                if (cookie.substring(0, 11) === 'XSRF-TOKEN=') {
+                    cookieValue = decodeURIComponent(cookie.substring(11));
+                    break;
+                }
+            }
+        }
+
+        return cookieValue;
```

Comment:
Is a nicer cookie handling API something that's worth including in the base Snowboard API?

Reply:
I've added a utility that wraps the [`js-cookie`](https://github.com/js-cookie/js-cookie) library, as it's super lightweight and has a much better API.